new nx create missing @eslint/eslintrc dependency

### Current Behavior

Created a brand new nx project, then imported into pnpm, then attempted build. Got error @eslint/eslintrc was missing which is true, it was not in package.json

### Expected Behavior

A brand new project should compile

### GitHub Repo

_No response_

### Steps to Reproduce

1. npx create-nx-workspace@latest (I did this through the Webstorm wizard for nx using the official nx plugin)
2. using node 20
3. for wizard:
4. Create React app with framework of choice
5. chose nextjs
6. integrated monorepo
7. picked my app name
8. No on app router
9. Yes on src
10. none on e2e framework
11. sass
12. github actions
13. At this point it installs with npm, creating my workspace
14. brought up nX console in Webstorm
15. click Refresh Workspace
16. Projects -> build
17. get immediate failure (Error: Cannot find module '../src/utils/find-workspace-root' shouldn't this also work??)
18. go to terminal on MacOS
19. cd into project folder
20. run command in terminal:  **pnpm import**
21. Delete node_modules from folder with Finder
22. run command in terminal:  **pnpm install**
23. run command: pnpm approve-builds
24. select all scripts and allow them to run (swc/core, sharp, nx)
25. go back to webstorm and nx console
26.  Projects -> build
27. this will actually successfully build at this point
28. Projects-> start
29. this will run, can go to localhost 3000 at this point
30. from terminal, run: nx reset
31. Projects -> build now this will fail, Cannot find package '@eslint/eslintrc' imported from (path)/eslint.config.mjs
32. From terminal, pnpm add @eslint/eslintrc -D
33. Projects -> build this now works
34. nx reset
35. now we get a different error,  Module @nx/react/plugins/jest in the transform option was not found.
at this point I gave up. not sure why nx reset and then nx build throws various errors



### Nx Report

```shell
⚠️ Unable to construct project graph.
Failed to process project graph.
     - apps/(projectname)/eslint.config.mjs: Error: Failed to load plugin '@next/next' declared in ' » eslint-config-next': Cannot find module '@next/eslint-plugin-next'
  Require stack:
  - /Volumes/dev/(projectname)/apps/(projectname)/__placeholder__.js
  Referenced from: /Volumes/dev/(projectname)/node_modules/.pnpm/eslint-config-next@14.2.16_eslint@9.19.0_typescript@5.7.3/node_modules/eslint-config-next/index.js
      at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
      at mod._resolveFilename (/Volumes/dev/(projectname)/node_modules/.pnpm/eslint-config-next@14.2.16_eslint@9.19.0_typescript@5.7.3/node_modules/eslint-config-next/index.js:50:26)
      at Function.resolve (node:internal/modules/helpers:187:19)
      at Module.resolve (file:///Volumes/dev/(projectname)/node_modules/.pnpm/@eslint+eslintrc@3.2.0/node_modules/@eslint/eslintrc/lib/shared/relative-module-resolver.js:23:46)
      at ConfigArrayFactory._loadPlugin (file:///Volumes/dev/(projectname)/node_modules/.pnpm/@eslint+eslintrc@3.2.0/node_modules/@eslint/eslintrc/lib/config-array-factory.js:1068:33)
      at ConfigArrayFactory._loadExtendedPluginConfig (file:///Volumes/dev/(projectname)/node_modules/.pnpm/@eslint+eslintrc@3.2.0/node_modules/@eslint/eslintrc/lib/config-array-factory.js:887:29)
      at ConfigArrayFactory._loadExtends (file:///Volumes/dev/(projectname)/node_modules/.pnpm/@eslint+eslintrc@3.2.0/node_modules/@eslint/eslintrc/lib/config-array-factory.js:808:29)
      at ConfigArrayFactory._normalizeObjectConfigDataBody (file:///Volumes/dev/(projectname)/node_modules/.pnpm/@eslint+eslintrc@3.2.0/node_modules/@eslint/eslintrc/lib/config-array-factory.js:749:25)
      at _normalizeObjectConfigDataBody.next (<anonymous>)
      at ConfigArrayFactory._normalizeObjectConfigData (file:///Volumes/dev/(projectname)/node_modules/.pnpm/@eslint+eslintrc@3.2.0/node_modules/@eslint/eslintrc/lib/config-array-factory.js:694:20)
     - apps/(projectname)/jest.config.ts: Error: ● Validation Error:

    Module @nx/react/plugins/jest in the transform option was not found.
           <rootDir> is: /Volumes/dev/(projectname)/apps/(projectname)

    Configuration Documentation:
    https://jestjs.io/docs/configuration
```

### Failure Logs

```shell

```

### Package Manager Version

pnpm 10.1.0

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/eslint/src/generators/utils/eslint-file.spec.ts
+++ b/packages/eslint/src/generators/utils/eslint-file.spec.ts
@@ -120,6 +120,45 @@ describe('@nx/eslint:lint-file', () => {
       ]);
     });
 
+    it('should install necessary dependencies', () => {
+      // mock eslint version
+      jest.spyOn(devkitInternals, 'readModulePackageJson').mockReturnValue({
+        packageJson: { name: 'eslint', version: '9.0.0' },
+        path: '',
+      });
+      tree.write('eslint.config.cjs', 'module.exports = {};');
+      tree.write(
+        'apps/demo/eslint.config.cjs',
+        `const baseConfig = require("../../eslint.config.cjs");
+
+module.exports = [
+  ...baseConfig,
+  {
+    files: [
+      "**/*.ts",
+      "**/*.tsx",
+      "**/*.js",
+      "**/*.jsx"
+    ],
+    rules: {}
+  },
+];`
+      );
+
+      addExtendsToLintConfig(tree, 'apps/demo', {
+        name: 'plugin:playwright/recommend',
+        needCompatFixup: true,
+      });
+
+      expect(readJson(tree, 'package.json').devDependencies)
+        .toMatchInlineSnapshot(`
+        {
+          "@eslint/compat": "^1.1.1",
+          "@eslint/eslintrc": "^2.1.1",
+        }
+      `);
+    });
+
     it('should add extends to flat config', () => {
       tree.write('eslint.config.cjs', 'module.exports = {};');
       tree.write(

--- a/packages/eslint/src/generators/utils/eslint-file.ts
+++ b/packages/eslint/src/generators/utils/eslint-file.ts
@@ -20,7 +20,11 @@ import {
   useFlatConfig,
 } from '../../utils/flat-config';
 import { getInstalledEslintVersion } from '../../utils/version-utils';
-import { eslint9__eslintVersion, eslintCompat } from '../../utils/versions';
+import {
+  eslint9__eslintVersion,
+  eslintCompat,
+  eslintrcVersion,
+} from '../../utils/versions';
 import {
   addBlockToFlatConfigExport,
   addFlatCompatToFlatConfig,
@@ -493,13 +497,19 @@ export function addExtendsToLintConfig(
       return addDependenciesToPackageJson(
         tree,
         {},
-        { '@eslint/compat': eslintCompat },
+        { '@eslint/compat': eslintCompat, '@eslint/eslintrc': eslintrcVersion },
         undefined,
         true
       );
     }
 
-    return () => {};
+    return addDependenciesToPackageJson(
+      tree,
+      {},
+      { '@eslint/eslintrc': eslintrcVersion },
+      undefined,
+      true
+    );
   } else {
     const plugins = (Array.isArray(plugin) ? plugin : [plugin]).map((p) =>
       typeof p === 'string' ? p : p.name

